### ISSUE TITLE
Crash in UnplashExample sample

### ISSUE THREAD

**Issue by chapayGhub** [rechsteiner/Parchment#138]
<img width="939" alt="screen shot 2018-03-03 at 12 09 22" src="https://user-images.githubusercontent.com/878464/36932579-e86c2dbc-1edb-11e8-8ad4-4e05494f106a.png">

**Comment by chapayGhub** [comment]
May be this is a solution

```  swift
  func itemWidth(for pagingItem: T) -> CGFloat {
    if let width = widthCache[pagingItem] {
      return width
    } else {
      let width = widthForPagingItem?(pagingItem, false)
      width.flatMap({ widthCache[pagingItem] = $0  })    //CHANGED LINE  
      return width ?? options.estimatedItemWidth
    }
  }
```

And I Found another one issue in this sample:


``` swift
struct ImageItem: PagingItem, Hashable, Comparable {
  let index: Int
  let title: String
  let headerImage: UIImage
  let images: [UIImage]
  
  var hashValue: Int {
    //The sum of two Integer throw overflow exception in my simulator 
    //return title.hashValue + headerImage.hashValue
    
    // I think this fix is good for example app 
    return title.hashValue/2 + headerImage.hashValue/2   //CHANGED LINE  
  }
  ......
```

**Comment by rechsteiner** [comment]
Thanks for the report. I’ll take a look!

**Comment by rechsteiner** [comment]
Yeah, looks like the hashValue implementation causes an integer overflow on older devices. Added a fix for this in #140. Thanks for catching that 🙌
